WordPress/Classes/Services/MediaService.m

```diff
@@ -586,6 +654,14 @@ - (void)imageForMedia:(Media *)mediaInRandomContext
         }
         WPImageSource *imageSource = [WPImageSource sharedSource];
         void (^successBlock)(UIImage *) = ^(UIImage *image) {
+            // Check the media hasn't been deleted whilst we were loading.
+            if (!media || media.isDeleted) {
+                if (success) {
+                    success(nil);
```

Comment:
Looks like this would go ahead against the `_Nonnull` in `success:(nullable void (^)(UIImage * _Nonnull image))success`.